mbed_host_tests/host_tests_plugins/host_test_plugins.py

```diff
@@ -107,16 +108,36 @@ def print_plugin_char(self, char):
         stdout.flush()
         return True
 
-    def check_mount_point_ready(self, destination_disk, init_delay=0.2, loop_delay=0.25):
+    def check_mount_point_ready(self, destination_disk, init_delay=0.2, loop_delay=0.25, target_id=None):
         """! Waits until destination_disk is ready and can be accessed by e.g. copy commands
-
         @return True if mount point was ready in given time, False otherwise
-
         @param destination_disk Mount point (disk) which will be checked for readiness
         @param init_delay - Initial delay time before first access check
         @param loop_delay - polling delay for access check
         """
+
+        if target_id:
+            mbeds = mbed_lstools.create()
+            mbeds_by_tid = mbeds.list_mbeds_by_targetid()   # key: target_id, value mbedls_dict()
+
+            # Wait for mount point to appear with mbed-ls
+            # and if it does check if mount point for target_id changed
+            # If mount point changed, use new mount point and check if its ready (os.access)
+            new_destination_disk = destination_disk
+            for i in range(25): # 25x 200ms = 5sec
+                if target_id in mbeds_by_tid:
+                    if 'mount_point' in mbeds_by_tid[target_id]:
+                        new_destination_disk = mbeds_by_tid[target_id]['mount_point']
+                        break
+                sleep(200)
```

Comment:
```mbeds = mbed_lstools.create()``` should be done inside the loop. Otherwise it will loop on same data.
Like:
```
            mbeds = mbed_lstools.create()
            mbeds_by_tid = mbeds.list_mbeds_by_targetid()
            for i in range(25): # 25x 200ms = 5sec
                if target_id in mbeds_by_tid:
                    if 'mount_point' in mbeds_by_tid[target_id]:
                        new_destination_disk = mbeds_by_tid[target_id]['mount_point']
                        break
                sleep(to)
```